neuro_extras/image_builder.py

```diff
@@ -84,137 +89,103 @@ async def _gen() -> AsyncIterator[bytes]:
+        self.logger.info(f"Using {context_uri} as the build context")
 
-        logging.info(f"Using {context_uri} as the build context")
-
+        # upload (if needed) build context and platform registry auth info
         build_uri = self._generate_build_uri()
-        await self._client.storage.mkdir(build_uri, parents=True, exist_ok=True)
+        await self._client.storage.mkdir(build_uri, parents=True)
         if context_uri.scheme == "file":
             local_context_uri, context_uri = context_uri, build_uri / "context"
-            logger.info(f"Uploading {local_context_uri} to {context_uri}")
+            self.logger.info(f"Uploading {local_context_uri} to {context_uri}")
             subprocess = await asyncio.create_subprocess_exec(
-                "neuro", "cp", "--recursive", str(local_context_uri), str(context_uri)
+                "neuro",
+                "--disable-pypi-version-check",
+                "cp",
+                "--recursive",
+                str(local_context_uri),
+                str(context_uri),
             )
             return_code = await subprocess.wait()
             if return_code != 0:
                 raise click.ClickException("Uploading build context failed!")
 
         docker_config = await self.create_docker_config()
         docker_config_uri = build_uri / ".docker.config.json"
-        logger.debug(f"Uploading {docker_config_uri}")
+        self.logger.debug(f"Uploading {docker_config_uri}")
         await self.save_docker_config(docker_config, docker_config_uri)
 
-        logger.info("Submitting a builder job")
-        image_ref = self.parse_image_ref(image_uri_str)
-        builder_container = await self._create_builder_container(
-            docker_config_uri=docker_config_uri,
-            context_uri=context_uri,
-            dockerfile_path=dockerfile_path,
-            image_ref=image_ref,
-            use_cache=use_cache,
-            build_args=build_args,
-            volume=volume,
-            env=env,
-            job_preset=job_preset,
+        cache_image = neuro_api.RemoteImage(
+            name="layer-cache/cache",
+            owner=self._client.config.username,
+            registry=str(self._client.config.registry_url),
+            cluster_name=self._client.cluster_name,
+        )
+        cache_repo = self.parse_image_ref(str(cache_image))
+        cache_repo = re.sub(r":.*$", "", cache_repo)  # drop tag
+
+        # mount build context and Kaniko auth info
+        volumes += (
+            f"{docker_config_uri}:/kaniko/.docker/config.json:ro",
+            # context dir cannot be R/O if we want to mount secrets there
+            f"{context_uri}:{KANIKO_CONTEXT_PATH}:rw",
         )
-        # TODO: set proper tags
-        job = await self._client.jobs.run(builder_container, life_span=4 * 60 * 60)
-        logger.info(f"The builder job ID: {job.id}")
-        return job
+        build_tags += (f"kaniko-builds:{image_uri_str}",)
```

Comment:
Since tags can't contain `//`,  we could use the form of `registry.neu.ro/username/imagename:tag` instead of `image://clustername/username/imagename:tag`

Reply:
yeah I agree the format of tags is too reduced artificially

Reply:
the sequence of ":/" is not allowed..(
the pattern used is `JOB_TAG_PATTERN = r"\A[a-z](?:[-.:/]?[a-z0-9])*\Z"`

how hard will it be to change that? (@romasku , @asvetlov)
- do we need to perform some DB migration? 
- why do we have restrictions for job tags structure in general? I didn't found those tags on job pods in k8s.

Reply:
No DB migration is required. At the moment, tags are only used to filter jobs - they have no effect on the k8s pod. So I see do not see any technical limitations here. Still we need at least some restrictions (

Reply:
Would be good to allow any potential image name (for instance "image://cluster/user/image/name:tag123").

However, if there are no other use cases for tags, but filtering, why don't we allow every possible combination of letters, digits, and characters "-.:/\_" ?
Smth like `\A(?:[-.:/\_a-zA-Z0-9])*\Z` might fit (if I'm not mistaken).

WDYT, @romasku , @artemlops?

Reply:
>  ... every possible combination of letters, digits, and characters "-.:/_" 

Sounds good to me. I'll discuss it during daily standup with other members of the team (just in case I am missing something) -- if they agree I will update the validation.

Reply:
awesome, thank you!

Reply:
We decided that it is OK to decrease restrictions here. How urgent is it? It's a tiny change but still takes some time.

Reply:
well, it would be good to have it asap (we wanna release neuro-extras with all bumped deps + inheritance of reconnection due to Synthesis feedback).

If the regex `\A(?:[-.:/\_a-zA-Z0-9])*\Z` is wrong - give me a hint of what's wrong and I will post a PR to replace it (and fix tests accordingly).

Reply:
Created a PR https://github.com/neuro-inc/platform-api/pull/1531. Will release it to staging soon after PR is merged.

Reply:
since a new version of platform API is already deployed, I've changed the tags to display full image URI. 
please, check it out, wdyt? @artemlops

Reply:
Perfect, I'm glad we implemented the full URIs